### PR TITLE
Ons tab karma test

### PR DESCRIPTION
**Pull request by andipavllo**

### CODE REVIEW

File: core/elements/ons-tab.spec.es6 [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      expect(element.children[1].classList.contains('tab-bar--fuga__button')).to.be.true;
+    });
+  });
+
+  describe('persistent attribute', () => {
+    it('adds a persistent state to the tab', function() {
+      let tabbar = new OnsTabbarElement();
+
+      tabbar.appendChild(element);
+      document.body.appendChild(tabbar);
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: This element is added to the DOM but never detached.

File: core/elements/ons-tab.spec.es6 [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      expect(element.isPersistent()).not.to.be.true;
+    });
+  });
+
+  describe('no-reload attribute', () => {
+    it('sets the tab as no-reloadable', function() {
+      let tabbar = new OnsTabbarElement();
+
+      tabbar.appendChild(element);
+      document.body.appendChild(tabbar);
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: This one too

File: core/elements/ons-tab.spec.es6 [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+      expect(element.canReload()).to.be.true;
+    });
+  });
+
+  describe('active attribute', () => {
+    it('sets whether a tab should be active or not', () => {
+      expect(element.hasAttribute('active')).not.to.be.true;
+
+      let tabbar = new OnsTabbarElement();
+      document.body.appendChild(tabbar);
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: This

File: core/elements/ons-tab.spec.es6 [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+      element._loadPageElement(myFunction);
+    });
+  });
+
+  describe('#setActive()', () => {
+    it('will set the tab as active', () => {
+      let tabbar = new OnsTabbarElement();
+
+      document.body.appendChild(tabbar);
+      document.body.innerHTML += '<ons-template id="page1"><ons-page></ons-page></ons-template><ons-template id="page2"><ons-page></ons-page></ons-template>';
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: If you add elements like this you can't remove them easily. Please create with `ons._util.createElement` and then remove them after the test is finished.

File: core/elements/ons-tab.spec.es6 [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+      expect(() => element._ensureElementPosition()).to.throw('This ons-tab element is must be child of ons-tabbar element.');
+
+      var parent = new OnsTabbarElement();
+      parent.appendChild(element);
+      expect(() => element._ensureElementPosition()).not.to.throw('This ons-tab element is must be child of ons-tabbar element.');
+    });
+  });
+
+  describe('_hasDefaultTemplate property', () => {
+    it('is, by defaut, true', () => {
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: Typo

File: core/elements/ons-tab.spec.es6 [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+      expect(document.getElementsByClassName('tab-bar__label')[0].innerHTML).to.equal('text');
+
+      element.setAttribute('label', 'new text');
+      expect(document.getElementsByClassName('tab-bar__label')[0].innerHTML).to.equal('new text');
+      expect(document.getElementsByClassName('tab-bar__label')[0].innerHTML).not.to.equal('text');
+    });
+  });
+
+  describe('class', () => {
+    it('has defaut properties', () => {
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: This test is difficult to understand.

File: core/elements/ons-tab.spec.es6 [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+      expect(element.hasAttribute('active')).to.be.true;
+    });
+  });
+
+  describe('icon attribute', () => {
+    it('sets icon name for the tab', () => {
+      let tabbar = new OnsTabbarElement();
+
+      tabbar.appendChild(element);
+      document.body.appendChild(tabbar);
```
Comment (argelius) [inline core/elements/ons-tab.spec.es6]: Clean up this element.

**Comment by andipavllo** [comment]
@argelius I applied the changes you suggested

**Comment by argelius** [comment]
Thanks @andipavllo !

**Pull request merged**